azkaban-common/src/test/java/azkaban/utils/UtilsTest.java

```diff
@@ -60,12 +60,12 @@ public void testValidCronExpressionV() {
 
     final DateTimeZone timezone = DateTimeZone.getDefault();
     Assert.assertTrue(Utils.isCronExpressionValid("0 0 3 ? * *", timezone));
-    Assert.assertTrue(Utils.isCronExpressionValid("0 0 3 ? * * 2017", timezone));
+    Assert.assertTrue(Utils.isCronExpressionValid("0 0 3 ? * * 2018", timezone));
```

Comment:
can we use something like getCurrentYear instead of hard coded year?

Reply:
I think it depends on the intention of this test.

The more likely culprit I think is the use of new Date() in 

https://github.com/HappyRay/azkaban/blob/b13ecc2854c0f28ba731fffb1586679329ed47ac/az-core/src/main/java/azkaban/utils/Utils.java#L617
cronExecutionTime.getNextValidTimeAfter(new Date()) == null)

This call doesn't allow the time to be controlled in the tests.

Reply:
We have to fix it again at the same time next year. Should we use a high number replacement? like "0 0 3 ? * * 2022".